test/RewardsStreamerMP.t.sol

```diff
@@ -2915,301 +2953,385 @@ contract CompoundTest is RewardsStreamerMPTest {
+        CheckVaultParams storage expectedAccountParams = expectedAccountState[account];
+        uint256 calcLockEnd = Math.max(
+            expectedVaultLockState[expectedAccountParams.account].lockEnd, vm.getBlockTimestamp()
+        ) + lockUpPeriod;
+        uint256 calcLockUpPeriod = calcLockEnd - vm.getBlockTimestamp(); //increased lock + remaining current lock
+        if (lockUpPeriod == 0 || (lockUpPeriod >= MIN_LOCKUP_PERIOD && lockUpPeriod <= MAX_LOCKUP_PERIOD)) {
+            //valid raw input
+            if (expectedVaultLockState[expectedAccountParams.account].totalLockUp + lockUpPeriod > MAX_LOCKUP_PERIOD) {
+                // but total lock time surpassed the maximum allowed
+                expectedRevert = StakeMath.StakeMath__AbsoluteMaxMPOverflow.selector;
+            } else {
+                expectedRevert = NO_REVERT;
+                expectedAccountParams.account = vaults[account];
+                uint256 expectedBonusMP = _bonusMP(stakeAmount, calcLockUpPeriod);
+                uint256 expectedMaxTotalMP = _maxTotalMP(stakeAmount, calcLockUpPeriod);
+                if (expectedVaultLockState[expectedAccountParams.account].lockEnd > vm.getBlockTimestamp()) {
+                    // in case stake increased for a locked vault
+                    // increases MP for the previous balance
+                    expectedBonusMP += _bonusMP(expectedAccountParams.stakedBalance, lockUpPeriod);
+                    expectedMaxTotalMP += _maxTotalMP(expectedAccountParams.stakedBalance, lockUpPeriod);
+                }
+
+                if (lockUpPeriod > 0) {
+                    //update lockup end
+                    expectedVaultLockState[expectedAccountParams.account].totalLockUp += lockUpPeriod;
+                    expectedVaultLockState[expectedAccountParams.account].lockEnd = calcLockEnd;
+                }
+                expectedAccountParams.stakedBalance = stakeAmount;
+                expectedAccountParams.vaultBalance = stakeAmount;
+                expectedSystemState.stakingBalance += stakeAmount;
+                expectedSystemState.totalStaked += stakeAmount;
+                expectedAccountParams.mpStaked = stakeAmount + expectedBonusMP;
+                expectedSystemState.totalMPStaked += stakeAmount + expectedBonusMP;
+                expectedAccountParams.mpAccrued = stakeAmount + expectedBonusMP;
+                expectedSystemState.totalMPAccrued += stakeAmount + expectedBonusMP;
+                expectedAccountParams.maxMP = expectedMaxTotalMP;
+                expectedSystemState.totalMaxMP += expectedMaxTotalMP;
+            }
+        } else {
+            if (lockUpPeriod > MAX_LOCKUP_PERIOD) {
+                expectedRevert = IStakeManager.StakingManager__InvalidLockPeriod.selector;
+            } else {
+                expectedRevert = StakeMath.StakeMath__InvalidLockingPeriod.selector;
+            }
+            return;
+        }
+    }
```

Comment:
These hard to parse blocks are essentially the same I've pointed out in https://github.com/vacp2p/staking-reward-streamer/pull/170#discussion_r1989011647 no?

What's the plan on getting those simplified.

As mentioned the last time, I think we should group the revert fuzz tests and extract them out.
It's otherwise very hard to follow what this is doing.